ui/src/app/pages/admin/admin-user-common.tsx

```diff
@@ -181,6 +188,45 @@ export const displayModuleExpirationDate = (
     getNullStringForExpirationDate(moduleName)
   );
 
+const isModuleRequiredByAccessTier = (
+  profile: Profile,
+  moduleName: AccessModule,
+  accessTierShortName: AccessTierShortNames
+): boolean => {
+  const tierEligibility = profile.tierEligibilities.find(
+    (tier) => tier.accessTierShortName === accessTierShortName
+  );
+  // Handling only ERACommon module.
+  return switchCase(moduleName, [
+    AccessModule.ERACOMMONS,
+    () =>
+      getAccessModuleConfig(moduleName).isEnabledInEnvironment &&
+      tierEligibility?.eraRequired,
+  ]);
+};
+
+export const displayTierBadgeByRequiredModule = (
+  profile: Profile,
+  moduleName: AccessModule
+) => {
+  return (
+    <div>
+      {(getAccessModuleConfig(moduleName)?.isRequiredByRT ||
```

Comment:
what do you think of this instead?
```
moduleName === ERA_COMMONS 
  ? isEraRequiredForTier(profile, AccessTierShortNames.Registered)
  : getAccessModuleConfig(moduleName)?.isRequiredByRT
```

Reply:
Yes, it looks cleaner and better. I'll update the code.